pymeasure/instruments/scientificmagnetics/scimag.py

```diff
@@ -0,0 +1,448 @@
+        map_values=True
+    )
+
+    terminal_voltage_limit = Instrument.setting(
+        "Y%04.1f",
+        """Sets the terminal voltage limit to nn.n V.""",
+        validator=strict_range,
+        values=[0, _VOLTAGE_LIMIT_MAX]
+    )
+
+    heater_current_output = Instrument.setting(
+        "W%03d",
+        """Sets the Heater current output to nnn mA.""",
+        validator=strict_range,
+        values=[0, 255]
+    )
+
+    calibrate_tesla_per_amp = Instrument.setting(
+        "C%08.6f",
+        """Calibrate the output to Tesla per Amp value of the superconducting magnet.
+         The Value is assumed to lie between 0.01 and 0.5 T/A
+         and will be set to zero if a number outside this range is entered.
+         """,
+        validator=strict_range,
+        values=[0.01, 0.5],
+    )
+
+    lower_setpoint = Instrument.setting(
+        "L%08.4f",
+        """Sets the lower setpoint. 
+        Unit depends on the unit set in the magnet controller.
+        Unit can be "T" or "A" and must be set before.
+        """,
+        validator=strict_range,
+        values=[0, 200]
+    )
+
+    upper_setpoint = Instrument.setting(
+        "U%08.4f",
+        """Sets the upper setpoint. 
+        Unit depends on the unit set in the magnet controller.
+        Unit can be "T" or "A" and must be set before.
+        """,
+        validator=strict_range,
+        values=[0, 200]
+    )
+
+    output_parameters_amps = Instrument.measurement(
+        "G",
+        """Returns dict of output parameters.""",
+        preprocess_reply=extract_value_output_parameters_amps
+    )
+
+    output_parameters_tesla = Instrument.measurement(
+        "N",
+        """Returns dict of output parameters.""",
+        preprocess_reply=extract_value_output_parameters_tesla
+    )
+
+    magnet_status = Instrument.measurement(
+        "J",
+        "Returns dict of magnet status.",
+        preprocess_reply=extract_value_magnet_status
+    )
+
+    current_status = Instrument.measurement(
+        "K",
+        "Returns dict of current status.",
+        preprocess_reply=extract_value_current_status
+    )
+
+    operating_parameters = Instrument.measurement(
+        "O",
+        "Returns dict of operation parameters",
+        preprocess_reply=extract_value_operating_parameters
+    )
+
+    set_point_status = Instrument.measurement(
+        "S",
+        "Returns dict of set point status.",
+        preprocess_reply=extract_value_set_point_status
+    )
+
+    def __init__(self, resourceName,
+                 ramp_rate_max=5,
+                 voltage_limit_max=1,
+                 field_max=10,
+                 **kwargs):
+        super().__init__(
+            resourceName,
+            "Scientific Magnetics SMC",
+            includeSCPI=False,
+            **kwargs
+        )
+        self._RAMP_RATE_MAX = ramp_rate_max  # A/s
+        self._VOLTAGE_LIMIT_MAX = voltage_limit_max  # V
+        self._T_MAX = field_max  # T
```

Comment:
Please not that updating these variables doesn't have any impact on the validators where you use these variables (see  #250, and pull requestes #371 & #376)